numpy/core/src/npysort/radixsort.c.src

```diff
@@ -0,0 +1,188 @@
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+
+#include "npy_sort.h"
+#include "npysort_common.h"
+#include <stdlib.h>
+
+/*
+ *****************************************************************************
+ **                            INTEGER SORTS                                **
+ *****************************************************************************
+ */
+
+
+/**begin repeat
+ *
+ * #TYPE = BYTE, UBYTE, SHORT, USHORT, INT, UINT, LONG, ULONG,
+ *         LONGLONG, ULONGLONG#
+ * #suff = byte, ubyte, short, ushort, int, uint, long, ulong,
+ *         longlong, ulonglong#
+ * #type = npy_byte, npy_ubyte, npy_short, npy_ushort, npy_int,
+ *         npy_uint, npy_long, npy_ulong, npy_longlong, npy_ulonglong#
+ * #sign = 1, 0, 1, 0, 1, 0, 1, 0, 1, 0#
+ */
+
+#if @sign@ == 0
+    #define KEY_OF(x) x
+#else
+    #define KEY_OF(x) x ^ (((@type@)1) << (sizeof(@type@) - 1))
+#endif 
+
+int
+radixsort_@suff@(void *start, npy_intp num, void *NOT_USED)
+{
+    npy_intp cnt[256 * sizeof(@type@)];
+    npy_intp i, j, l;
+    @type@* arr;
+    @type@* aux;
+    npy_intp a[sizeof(@type@)];
+    npy_intp b[sizeof(@type@)];
+    
+    arr = (@type@*)start;
+    aux = malloc(num * sizeof(@type@));
+
+    if (aux == NULL)
+        return -NPY_ENOMEM;
+
+    for (i = 0; i < num; i++) {
+        @type@ k = KEY_OF(arr[i]);
+
+        for (l = 0; l < num; l++) {
```

Comment:
Is `num` the correct loop bound here? Shouldn't this be `sizeof(@type@)` instead?

Reply:
Fixed.